### ISSUE TITLE
STATIC_DRAW

### ISSUE THREAD

**Issue by dekdevy** [oframe/ogl#12]
https://github.com/oframe/ogl/blob/6ab86d133acd1aeb31239641f9c8de40941c7be3/src/core/Geometry.js#L85

This line makes the assumption that the buffer is altered rarely. In my application, when I change a buffer often, this will break after a few frames. Setting it to `DYNAMIC_DRAW` fixes the issue. 

The error that is thrown:
`GL ERROR :GL_INVALID_OPERATION : glDrawElements: attempt to access out of range vertices in attribute 3 `
Since changing the usage type fixes the issue for me, I am assuming that the reason lies therein. However, I am not 100% sure that this is the actual problem, so I cant speak with full certainty on this. It might after all be some other obscure bug.

**Comment by dekdevy** [comment]
Proposed solution: specifying a buffer usage mode on a per attribute basis, defaulting to either dynamic or static.

**Comment by dekdevy** [comment]
Actually, I looked into it a bit more and this seems to actually have been introduced when these changes were made: https://github.com/oframe/ogl/commit/183e2903f1974134b95e67fb3df24fbf7281207b#diff-a7251e9e389d43e7878065274e4d067dR82

So I am assuming that this is in fact not the issue, but related to the change.

**Comment by dekdevy** [comment]
https://github.com/oframe/ogl/blob/6ab86d133acd1aeb31239641f9c8de40941c7be3/src/core/Geometry.js#L81

I think this is the actual issue here, and I have no idea why the draw mode seems to have any impact on the problem. If I remove this condition and always bind the buffer - like it was before the change - everything seems to be back to normal. So I am assuming that OGL at some point binds a buffer without setting `glState.boundBuffer` accordingly, causing the condition to result in false, which in return causes the attribute to update some other buffer. Ill look into that some more.

**Comment by gordonnl** [comment]
Hey @dekdevy, sorry I'm late at getting to this.
Do you have a simple example where I can see this occurring to test some solutions?
I'd be totally up for adding an ```attr.usage``` property for ```DYNAMIC_DRAW``` if it makes a difference.

**Comment by dekdevy** [comment]
Hey @gordonnl, I managed to find the error, and it was on my side. I am maintaining my own fork of OGL, and during the merging of your change to the attribute update function, I failed to properly implement one  line, which caused `this.glState.boundBuffer` to contain a wrong value, and consequently mess with the buffers. So it is all good and OGL is working flawlessly now. The ability to change the draw mode might still be useful, although I am not sure about the impact or proper usage.
